src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -768,6 +782,93 @@ CHIP_ERROR SilabsTracer::SplitNamedTraceString(CharSpan appOperationKey, CharSpa
     return CHIP_NO_ERROR;
 }
 
+#if defined(configGENERATE_RUN_TIME_STATS) && configGENERATE_RUN_TIME_STATS == 1
+
+CHIP_ERROR SilabsTracer::OutputTaskStatistics()
+{
+    VerifyOrReturnError(isLogInitialized(), CHIP_ERROR_UNINITIALIZED);
+
+    UBaseType_t uxArraySize = uxTaskGetNumberOfTasks();
```

Comment:
I see you are using all freeRTOS API. Which you might need for the stats. I don't know if the abstraction exist with CMSIOS.
However to get the task list and task array CMSIS OS does have the API and is quite clean i find.
You have an example on ` examples/platform/silabs/MemMonitoring.cpp`